src/test/java/com/google/firebase/auth/FirebaseUserManagerTest.java

```diff
@@ -1846,6 +1956,53 @@ public void testCreateSamlProvider() throws Exception {
     assertEquals("https://projectId.firebaseapp.com/__/auth/handler", spConfig.get("callbackUri"));
   }
 
+  @Test
+  public void testCreateSamlProviderAsync() throws Exception {
+    TestResponseInterceptor interceptor = initializeAppForUserManagement(
+        TestUtils.loadResource("saml.json"));
+    SamlProviderConfig.CreateRequest createRequest =
+        new SamlProviderConfig.CreateRequest()
+          .setProviderId("saml.provider-id")
+          .setDisplayName("DISPLAY_NAME")
+          .setEnabled(true)
+          .setIdpEntityId("IDP_ENTITY_ID")
+          .setSsoUrl("https://example.com/login")
+          .addX509Certificate("certificate1")
+          .addX509Certificate("certificate2")
+          .setRpEntityId("RP_ENTITY_ID")
+          .setCallbackUrl("https://projectId.firebaseapp.com/__/auth/handler");
+
+    SamlProviderConfig config =
+        FirebaseAuth.getInstance().createSamlProviderConfigAsync(createRequest).get();
+
+    checkSamlProviderConfig(config, "saml.provider-id");
+    checkRequestHeaders(interceptor);
+    checkUrl(interceptor, "POST", PROJECT_BASE_URL + "/inboundSamlConfigs");
+    GenericUrl url = interceptor.getResponse().getRequest().getUrl();
+    assertEquals("saml.provider-id", url.getFirst("inboundSamlConfigId"));
+
+    GenericJson parsed = parseRequestContent(interceptor);
+    assertEquals("DISPLAY_NAME", parsed.get("displayName"));
+    assertTrue((boolean) parsed.get("enabled"));
+
+    Map<String, Object> idpConfig = (Map<String, Object>) parsed.get("idpConfig");
```

Comment:
Can you use `GenericJson` here instead of `Map<String, Object>`? If possible, that might clean up lot of the generic warnings that result from these casts.

Reply:
I think you've suggested the same thing before, but when I try to cast to GenericJson I get compile errors:

```java.lang.ClassCastException: com.google.api.client.util.ArrayMap cannot be cast to com.google.api.client.json.GenericJson
	at com.google.firebase.auth.FirebaseUserManagerTest.testCreateSamlProviderAsync(FirebaseUserManagerTest.java:1988)```